Backend.php

```diff
@@ -0,0 +1,26 @@
+<?
+
+namespace iFixit\TokenBucket;
+
+require_once __DIR__ . DIRECTORY_SEPARATOR . 'StoredBucket.php';
+
+interface Backend {
+   /**
+    * Value for when there was no key found in storage.
+    */
+   const MISS = null;
+
+   /**
+    * Retrieve the value of the given string key.
+    * Needs to return an StoredBucket.
+    *
+    * should return a StoredBucket
+    */
+   public function get($key);
+   /**
+    * Set the value of the given string key to whatever type value is.
+    *
+    * Expiration time is given as an integer number of seconds.
```

Comment:
Oh wait... nevermind, this is expiration. Hmm.. this could actually be calculated from the bucket right? i.e. you'd want it to expire no sooner than when the bucket would end up being full again.